### ISSUE TITLE
clang-10 PETSc/Trilinos warnings

### ISSUE THREAD

**Issue by tjhei** [dealii/dealii#10400]
Not sure why I didn't see them before, but I am getting
```
In file included from ../source/numerics/dof_output_operator.cc:24:
In file included from ../include/deal.II/lac/petsc_block_vector.h:27:
In file included from ../include/deal.II/lac/petsc_vector.h:28:
../include/deal.II/lac/petsc_vector_base.h:110:7: warning: definition of implicit copy constructor for 'VectorReference' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]
      operator=(const VectorReference &r) const;
      ^
../include/deal.II/lac/petsc_vector_base.h:1081:12: note: in implicit copy constructor for 'dealii::PETScWrappers::internal::VectorReference' first required here
    return internal::VectorReference(*this, index);
           ^
In file included from ../source/numerics/dof_output_operator.cc:26:
In file included from ../include/deal.II/lac/trilinos_parallel_block_vector.h:27:
../include/deal.II/lac/trilinos_vector.h:122:7: warning: definition of implicit copy constructor for 'VectorReference' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]
      operator=(const VectorReference &r) const;
      ^
../include/deal.II/lac/trilinos_vector.h:1477:14: note: in implicit copy constructor for 'dealii::TrilinosWrappers::internal::VectorReference' first required here
      return internal::VectorReference(*this, index);
             ^
2 warnings generated.
```

I don't have time to look at them right now...

**Comment by tamiko** [comment]
@tjhei Also see https://cdash.43-1.org/viewBuildError.php?type=1&buildid=6843
You didn't see warnings before because you probably compiled with "-std=c++17" before but we now default to whatever the compiler sets as default and this is `-std=c++14`.

**Comment by drwells** [comment]
We fixed these last year - see #10403.
